src/pkg/utils/network/network.go

```diff
@@ -0,0 +1,23 @@
+package network
+
+import (
+	"net/url"
+
+	"github.com/defenseunicorns/zarf/src/pkg/transform"
+)
+
+// DoHostnamesMatch returns a boolean indicating if the hostname of two different URLs are the same.
+func DoHostnamesMatch(logger transform.Log, url1 string, url2 string) (bool, error) {
+	parsedURL1, err := url.Parse(url1)
+	if err != nil {
+		logger("unable to parse the url (%s)", url1)
+		return false, err
+	}
+	parsedURL2, err := url.Parse(url2)
+	if err != nil {
+		logger("unable to parse the url (%s)", url2)
+		return false, err
+	}
+
+	return parsedURL1.Hostname() == parsedURL2.Hostname(), nil
+}
```

Comment:
Because we are already returning any errors from `url.Parse()` and logging a message for it, we can return a wrapped error with the message rather than use a logger.

```suggestion
// DoHostnamesMatch returns a boolean indicating if the hostname of two different URLs are the same.
func DoHostnamesMatch(url1 string, url2 string) (bool, error) {
	parsedURL1, err := url.Parse(url1)
	if err != nil {
		return false, fmt.Errorf("unable to parse the url (%s): %w", url1, err)
	}
	parsedURL2, err := url.Parse(url2)
	if err != nil {
		return false, fmt.Errorf("unable to parse the url (%s): %w", url2, err)
	}

	return parsedURL1.Hostname() == parsedURL2.Hostname(), nil
}
```